src/server/scripts/Kalimdor/zone_orgrimmar.cpp

```diff
@@ -263,8 +264,563 @@ class npc_thrall_warchief : public CreatureScript
+                        break;
+                    case EVENT_HERALD_SCENE10:
+                        Talk(SAY_THRALL_6);
+                        events.ScheduleEvent(EVENT_HERALD_SCENE11, 9000);
+                        break;
+                    case EVENT_HERALD_SCENE11:
+                        if (Creature* jaina = ObjectAccessor::GetCreature(*me, jainaGUID))
+                            jaina->AI()->Talk(SAY_JAINA_1);
+                        events.ScheduleEvent(EVENT_HERALD_SCENE12, 6000);
+                        break;
+                    case EVENT_HERALD_SCENE12:
+                        if (Creature* jaina = ObjectAccessor::GetCreature(*me, jainaGUID))
+                            jaina->AI()->Talk(SAY_JAINA_2);
+                        events.ScheduleEvent(EVENT_HERALD_SCENE13, 14000);
+                        break;
+                    case EVENT_HERALD_SCENE13:
+                        if (Creature* jaina = ObjectAccessor::GetCreature(*me, jainaGUID))
+                            jaina->AI()->Talk(SAY_JAINA_3);
+                        events.ScheduleEvent(EVENT_HERALD_SCENE14, 10000);
+                        break;
+                    case EVENT_HERALD_SCENE14:
+                        if (Creature* jaina = ObjectAccessor::GetCreature(*me, jainaGUID))
+                        {
+                            jaina->AI()->Talk(SAY_JAINA_4);
+                            jaina->GetMotionMaster()->MoveTargetedHome();
+                            jaina->DespawnOrUnsummon(5000);
+                        }
+                        events.ScheduleEvent(EVENT_HERALD_SCENE15, 7000);
+                        break;
+                    case EVENT_HERALD_SCENE15:
+                        me->GetMotionMaster()->MoveTargetedHome();
+                        Talk(SAY_THRALL_7);
+
+                        if (Creature* sylvanas = ObjectAccessor::GetCreature(*me, sylvanasGUID))
+                            sylvanas->GetMotionMaster()->MoveTargetedHome();
+
+                        if (Creature* portal = ObjectAccessor::GetCreature(*me, stormwindPortalGUID))
+                            portal->DespawnOrUnsummon();
+
+                        if (GameObject* leftPortal = ObjectAccessor::GetGameObject(*me, ucPortalGUIDs[0]))
+                            leftPortal->RemoveFromWorld();
+
+                        if (GameObject* rightPortal = ObjectAccessor::GetGameObject(*me, ucPortalGUIDs[1]))
+                            rightPortal->RemoveFlag(GAMEOBJECT_FLAGS, GO_FLAG_NOT_SELECTABLE);
+
+                        events.ScheduleEvent(EVENT_HERALD_RESET, 60000);
+                        break;
+                    case EVENT_HERALD_RESET:
+                        Initialize();
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+    private:
+        EventMap events;
+        ObjectGuid guardsGUIDs[4];
```

Comment:
please use a define or a const for the size of an array if you are going to use the size few times to ensure the correct size is always used